### PR TITLE
Fix: add values.schema.json validation

### PR DESCRIPTION
**Pull request by wojtekzyla**
# Description

This PR adds values.schema.json to validate `values.yaml` in helm chart

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [x] Refactor/improvement

## How Has This Been Tested?

Manual testing

## Checklist

- [x] My commit message is [conventional](https://www.conventionalcommits.org/en/v1.0.0/#summary)
- [x] I have run pre-commit on all files before creating the PR
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

File: charts/splunk-connect-for-snmp/values.schema.json [hunk 1]
```diff
@@ -0,0 +1,613 @@
...
+      "type": "object"
+    },
+    "mongodb": {
+      "type": "object"
+    },
+    "redis": {
+      "type": "object"
+    },
+    "mibserver": {
+      "type": "object"
```
Comment (ajasnosz) [inline charts/splunk-connect-for-snmp/values.schema.json:610]: I think I would just allow to configure repository, pullPolicy and tag for mibserver

**Review by ajasnosz — CHANGES_REQUESTED** [review]

File: charts/splunk-connect-for-snmp/values.schema.json [hunk 2]
```diff
@@ -0,0 +1,613 @@
...
+        },
+        "valuesFileName": {
+          "type": "string"
+        },
+        "keepSectionFiles": {
+          "type": "boolean"
+        }
+      }
+    },
+    "splunk": {
```
[the remark this answers is not in the document]
Comment (wojtekzyla) [reply charts/splunk-connect-for-snmp/values.schema.json:119]: I was thinking about it. The problem is that if someone wants to use only o11y without a splunk, then these are not required. However they are still provided in the values.yaml from charts as empty strings so we could change it `required`

File: charts/splunk-connect-for-snmp/values.schema.json [hunk 3]
```diff
@@ -0,0 +1,613 @@
...
+    "scheduler": {
+      "type": "object",
+      "additionalProperties": false,
+      "properties": {
+        "groups": {
+          "type": "string"
+        },
+        "profiles": {
+          "type": "string"
+        },
```
[the remark this answers is not in the document]
Comment (wojtekzyla) [reply charts/splunk-connect-for-snmp/values.schema.json:217]: The syntax we use here is a multiline string. The `|` sign indicates the beginning of the multiline string ([helm documentation](https://helm.sh/docs/chart_template_guide/yaml_techniques/#strings-in-yaml)). This string is the pasted into the `config.yaml` file in configmap

```
profiles: |
    smart_profile:
      frequency: 100
      condition:
        type: field
        field: "SNMPv2-MIB.sysDescr"
        patterns:
          - '.*linux.*'
      varBinds:
        - ['SNMPv2-MIB']
        - ['SNMPv2-MIB', 'sysName']
        - ['SNMPv2-MIB', 'sysUpTime',0]
```

**Review by ajasnosz — CHANGES_REQUESTED** [review]

File: examples/lightweight_installation.yaml [hunk 4]
```diff
@@ -1,8 +1,8 @@
 splunk:
   enabled: true
   protocol: https
-  host: ###SPLUNK_HOST###
-  token: ###SPLUNK_TOKEN###
+  host: ""###SPLUNK_HOST###"
+  token: ""###SPLUNK_TOKEN###"
```
[the remark this answers is not in the document]
Comment (wojtekzyla) [reply examples/lightweight_installation.yaml]: fixed

**Review by ajasnosz — APPROVED** [review]
Add changelog

**Pull request merged**